File: algo/src/main/java/com/hartwig/actin/algo/evaluation/molecular/GeneIsOverexpressed.java
package com.hartwig.actin.algo.evaluation.molecular;

import com.hartwig.actin.PatientRecord;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.datamodel.EvaluationResult;
import com.hartwig.actin.algo.datamodel.ImmutableEvaluation;
import com.hartwig.actin.algo.evaluation.EvaluationFunction;

import org.jetbrains.annotations.NotNull;

public class GeneIsOverexpressed implements EvaluationFunction {

    GeneIsOverexpressed() {
    }

    @NotNull
    @Override
    public Evaluation evaluate(@NotNull PatientRecord record) {
        return ImmutableEvaluation.builder().result(EvaluationResult.FAIL).addFailMessages("No expression data available").build();
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/molecular/GeneIsOverexpressed.kt
package com.hartwig.actin.algo.evaluation.molecular

import com.hartwig.actin.algo.evaluation.EvaluationFactory
import com.hartwig.actin.datamodel.algo.Evaluation
import com.hartwig.actin.datamodel.molecular.MolecularRecord
import java.time.LocalDate

class GeneIsOverexpressed(maxTestAge: LocalDate? = null) : MolecularEvaluationFunction(maxTestAge, false) {

    override fun evaluate(molecular: MolecularRecord): Evaluation {
        return EvaluationFactory.undetermined("Overexpression of genes in RNA undetermined", missingGenesForEvaluation = true)
    }
}